### ISSUE TITLE
Creating visualizations combining multiple tags

### ISSUE THREAD

**Issue by georgedahl** [tensorflow/tensorboard#597]
Currently, as I understand it, there is a one-to-one correspondence between TensorBoard visualizations (plots etc.) and summary tags in the events file. This forces the user to modify what gets written to the events file to change what plots (or other visualizations) get produced, even when the data necessary to make the desired plot already exists in the events file.

The only mechanism TensorBoard has for grouping multiple data series onto the same axes is to make them different runs with the exact same tag. However, this will often require a user to maintain multiple SummaryFileWriters which is inconvenient and must be done anew for every piece of experiment code (things in TF like tf.train.MonitoredSession do not make this easy by default).

As a motivating example of the types of things I want to do, I want to make a training curve for a single run of a TensorFlow program that is already started running and is taking a very long time. I want to compare a scalar summary with tag train_loss and a scalar summary with tag:validataion_loss on the same axes with global_step on the abscissa. I would like to tell the UI somehow to merge the two plots it already creates. Another thing I want to do is to show these two tags across several runs of my program, but I want to have the color based on the run and the line style (or marker style) based on the tag. Both of these things are hard to do given that I have already started writing the events file.

Although I am most concerned about scalar summaries, this problem can arise with lots of other types of data, including image summaries. @dandelionmane might have a good example for image summaries where this type of configurable grouping would be useful.

Currently what I do is just abandon TensorBoard and write my own ipython notebook to read the events files and make the plots I want and I copy this for each new project. So this is a critical missing feature. Although it is possible to get part of the behavior I want by making multiple subfolders and using the same tag, I don't always know what curves I want to group among my scalar summaries when I am writing the code that writes the events files and even when I do this clutters up my training and eval code.

**Comment by lshe7842** [comment]
Here's a similar post: #300

**Comment by dandelionmane** [comment]
Yes, this quite related to #300. This is more of a general request, #300 suggests a specific implementation.

@chihuahua and I are planning to meet later today and discuss approaches for solving this issue; we'll keep @leandro-gracia-gil's suggested solution in mind. Thanks for linking it, @lshe7842.

**Comment by bileschi** [comment]
@psybuzz @stephanwlee is there more to say about this?

**Comment by psybuzz** [comment]
> Currently what I do is just abandon TensorBoard and write my own ipython notebook to read the events files and make the plots I want

Thanks for the report.  This sounds extremely inconvenient.  I don't have a plan to address this soon, but I'd like to keep it open and self-assign.

The Custom Scalars dashboard offers one way to address this, but I think there's a lot of friction involved in using it.  I'm not convinced that having users modify their Python programs with `summary_lib.custom_scalar_pb` or `summary_lib.custom_scalar` is a great solution.  We could investigate solutions in the TensorBoard frontend layer.
